apps/aecore/lib/aecore/peers/sync.ex

```diff
@@ -1,364 +1,607 @@
 defmodule Aecore.Peers.Sync do
   @moduledoc """
-  Contains peer sync functionality
+  This module is responsible for the Sync logic between Peers to share blocks between eachother
   """
 
   use GenServer
 
-  alias Aecore.Peers.Worker, as: Peers
-  alias Aehttpclient.Client, as: HttpClient
-  alias Aecore.Chain.Worker, as: Chain
+  alias __MODULE__
   alias Aecore.Chain.Header
-  alias Aecore.Tx.Pool.Worker, as: Pool
+  alias Aecore.Chain.Worker, as: Chain
   alias Aecore.Chain.BlockValidation
-  alias Aecore.Peers.PeerBlocksTask
-  alias Aecore.Chain.Header
-  alias Aecore.Chain.Block
+  alias Aecore.Persistence.Worker, as: Persistence
+  alias Aecore.Peers.PeerConnection
+  alias Aeutil.Scientific
+  alias Aecore.Tx.Pool.Worker, as: Pool
+  alias Aecore.Peers.Jobs
+  alias Aecore.Peers.Worker, as: Peers
+  alias Aecore.Peers.Events
 
   require Logger
 
-  @peers_target_count Application.get_env(:aecore, :peers)[:peers_target_count]
+  @typedoc "Structure of a peer to sync with"
```

Comment:
I'll leave it as is, because if I move it to the Peers module it will clash with the other Peer structure, for connecting the Peers. Maybe it could be changed in the future as the Sync will be improved.